File: frontend-app/src/core/model/dto/response/DispositivoResponse.ts
import type { DispositivoStatus } from "../types";
import { DEFAULT_AGENTE_OBJ, type AgenteResponse } from "./AgenteResponse";
import { DEFAULT_EMPRESA_OBJ, type EmpresaResponse } from "./EmpresaResponse";

export interface DispositivoResponse {
    id : number;
    host : string;
    nome : string;
    descricao : string;
    localizacao : string;
    sendoMonitorado : boolean;
    status : DispositivoStatus;
    latenciaMedia : number;
    stateAtualizadoEm : Date;
    monitoradoPorAgente : boolean;
    empresa : EmpresaResponse;
    agente : AgenteResponse;
}

export const DEFAULT_DISPOSITIVO_OBJ : DispositivoResponse = {
    id: 0,
    host: '',
    nome: '',
    descricao: '',
    localizacao: '',
    sendoMonitorado: false,
    status : 'INATIVO',
    latenciaMedia: 0,
    stateAtualizadoEm: new Date( 0 ),
    monitoradoPorAgente: false,
    empresa: DEFAULT_EMPRESA_OBJ,
    agente: DEFAULT_AGENTE_OBJ
};

File: main-api/src/main/java/com/italomonitor/main/dto/response/DispositivoResponse.java
package com.italomonitor.main.dto.response;

import java.util.Date;

import com.italomonitor.main.enums.DispositivoStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class DispositivoResponse {

    private Long id;
    
    private String host;
    
    private String nome;
    
    private String descricao;
    
    private String localizacao;
    
    private boolean sendoMonitorado;
    
    private DispositivoStatus status;
    
    private int latenciaMedia;
    
    private Date stateAtualizadoEm;
    
    private EmpresaResponse empresa;

}
